docs/source/configurator.rst

```diff
@@ -0,0 +1,173 @@
+
+The idea is the following: The components(nodes, carriage mechanisms, converters, re-sequencers...etc.) are agnostic
+to one another as long as the carriage mechanism is capable of transporting the payload the node expects. But their
+wiring and their factories are actually the configurator classes. The configurator class defines the type and
+name of the configuration options relevant to the component and wires it with its immediate dependencies. This
+way the factories that create the system are separate from the system's business logic and this separation is
+meant to be kept. There is a simple naming correspondence. For instance the business logic of the processing
+nodes are in the :py:mod:`ebu_tt_live.node` package while their configurators are to be found in the
+:py:mod:`ebu_tt_live.config.node` module.
+
+The configuration file structure
+--------------------------------
+
+The configuration file is JSON at the moment. The general structure looks like this:
+
+    ::
+
+        {
+            "nodes": {
+                "node1": {
+                    "type": "simple-producer",
+                    "output": { ... }
+                    ...
+                },
+                "node2": { ... },
+                ...
+            },
+            "backend": {
+                "type": "twisted",
+                ...
+            }
+        }
+
+The nodes are supposed to be compatible with the chosen backend. The backend object is instantiated first and
+every component configurator has access to the backend that was created and it is their responsibility to
+ensure that compatibility is checked with the backend. The backend provides a framework for
+timing things, network stack access, threading... etc. Different backends may offer different features and if
+a component configurator is not compatible with the backend object it should raise a ConfigurationError.
+
+The node contains either an input or an output section or both in case it is a node that both receives and emits
+sequences at the same time, like a forwarder node, a handover node, a converter... etc. The input and output bits
+define the carriage mechanism that that side is supposed to be talking to. So providing 2 nodes: 1) a producer with
+an output carriage mechanism that is "websocket" and 2) a consumer of an input carriage mechanism of the same websocket
+address that the producer is using essentially creates a loopback through the network stack between the 2 nodes
+and they will talk to one another. An even more convenient way of doing the same thing is to use
+the carriage mechanism type: ``direct`` which simply transfers the payload data in memory without going via e.g. a network stack.
+
+The more detailed options are: ::
+
+    nodes
+    ├─[nodeN] : a node to configure - any from node1 to node9
+    │ ├─type : ["simple-consumer" | "simple-producer" | "resequencer" | "ebuttd-encoder" | "buffer-delay" | "retiming-delay"]
```

Comment:
There is no such node. You either want a distributing one a handover one or a consumer one